### PR TITLE
Parallel tests in CI

### PR DESCRIPTION
**Pull request by ArturGajowy**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

This shortens the full build (unit tests + integration included) down to 45 mins, with the tests starting after just 6 minutes (!) That's much faster feedback for all our work, also before bors is involved.

See commit log for details why certain things are changed/removed.

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -42,43 +43,38 @@ pipeline:
...
-  run-unit-tests:
-    <<: *sbtenv
-    commands:
-      - sbt test it:test
-
-  run-code-analysis:
-    <<: *sbtenv
-    commands:
-      - sbt scalafmtCheck coverage
-      - ./scripts/ci/codecov-upload
```
Comment (KentShikama) [inline .drone.yml]: So we no longer run code coverage? I'm not against it if so, I'm just wondering.

File: .drone.yml [hunk 2]
```diff
@@ -42,43 +43,38 @@ pipeline:
...
       event: [ push, tag ]
 
+  # pr
+
+  run-unit-tests:
+    group: parallel
+    <<: *sbtenv
+    commands:
+      - cp -r . ../step.run-unit-tests && cd ../step.run-unit-tests
+      - sbt test it:test
```
Comment (KentShikama) [inline .drone.yml]: sbt it:test runs unit tests?

File: .drone.yml (hunk 1 above)
Comment (ArturGajowy) [reply .drone.yml]: See the commit message and/or discord. We haven't been running it for 2 months already, due to some kind of bug/misconfig. It's out of this PRs scope to fix that. This change is merely a clean up making the status quo explicit :)

File: .drone.yml (hunk 2 above)
Comment (ArturGajowy) [reply .drone.yml]: I'm not sure. Does it? I'd leave it as is so that nobody has to wonder about that :)

File: .drone.yml (hunk 1 above)
Comment (KentShikama) [reply .drone.yml]: I like how descriptive your commit messages are.

File: .drone.yml (hunk 2 above)
Comment (lukasz-golebiewski) [reply .drone.yml]: I'd say this should run together with integration tests. Doesn't need to change within this PR though

Comment (ArturGajowy) [reply .drone.yml]: I'm not even sure what `it:test` is :sweat_smile:

File: .drone.yml [hunk 3]
```diff
@@ -42,43 +43,38 @@ pipeline:
...
       - /var/cache/rchain-build/.sbt:/root/.sbt
       - /var/cache/rchain-build/.ivy2:/root/.ivy2
       - /var/run/docker.sock:/var/run/docker.sock
     when:
       event: [ push, tag ]
 
+  # pr
+
+  run-unit-tests:
+    group: parallel
```
Comment (lukasz-golebiewski) [inline .drone.yml]: How does this parallelize the tests?

Comment (ArturGajowy) [reply .drone.yml]: https://0-8-0.docs.drone.io/pipelines, see Parallel Execution

File: .drone.yml (hunk 2 above)
Comment (KentShikama) [reply .drone.yml]: I gave it a run 

```
$ sbt it:test
[info] InMemoryStoreConcurrencyTests:
[info] CORE-589 produce and consume in a multithreaded fashion
Finished InMemory: 500 iterations.
Time per thread: avg: 7309.80 ms; min: 7292.59 ms; max: 7332.91 ms
[info] - should show high performance (11 seconds, 84 milliseconds)
[info] produce and consume with monix.Tasks
Finished InMemory: 500 iterations.
Time per thread: avg: 955.43 ms; min: 780.50 ms; max: 1207.45 ms
[info] - should work (5 seconds, 679 milliseconds)
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.lmdbjava.ByteBufferProxy$AbstractByteBufferProxy (file:/Users/kent/.ivy2/cache/org.lmdbjava/lmdbjava/jars/lmdbjava-0.6.1.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of org.lmdbjava.ByteBufferProxy$AbstractByteBufferProxy
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[info] LMDBStoreConcurrencyTestsWithTls:
[info] CORE-589 produce and consume in a multithreaded fashion
Finished LMDB with TLS: 500 iterations.
Time per thread: avg: 25448.36 ms; min: 25345.17 ms; max: 25493.15 ms
[info] - should show high performance (28 seconds, 316 milliseconds)
[info] produce and consume with monix.Tasks
Finished LMDB with TLS: 500 iterations.
Time per thread: avg: 3695.90 ms; min: 3139.30 ms; max: 4172.58 ms
[info] - should work (19 seconds, 963 milliseconds)
[info] LMDBStoreConcurrencyTestsNoTls:
[info] CORE-589 produce and consume in a multithreaded fashion
Finished LMDB NO_TLS: 500 iterations.
Time per thread: avg: 18583.58 ms; min: 18559.59 ms; max: 18604.44 ms
[info] - should show high performance (18 seconds, 806 milliseconds)
[info] produce and consume with monix.Tasks
Finished LMDB NO_TLS: 500 iterations.
Time per thread: avg: 4004.09 ms; min: 3999.59 ms; max: 4008.89 ms
[info] - should work (20 seconds, 342 milliseconds)
[info] ScalaTest
[info] Run completed in 1 minute, 56 seconds.
[info] Total number of tests run: 6
[info] Suites: completed 3, aborted 0
[info] Tests: succeeded 6, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 6, Failed 0, Errors 0, Passed 6
[success] Total time: 156 s, completed Dec 14, 2018, 1:15:10 AM
```

Comment (KentShikama) [reply .drone.yml]: I see; yeah I agree with @lukasz-golebiewski that this should go together with integration tests in the future

Comment (ArturGajowy) [reply .drone.yml]: So it doesn't run unit tests. Having this moved to the integration-test step might allow squeezing another 2-3 mins due to better parallelization. Let's keep it for latter PRs though :)

**Pull request merged**